frontend/src/pulldasher/pulls-context.tsx

```diff
@@ -26,8 +28,25 @@ export function useSetFilter(): FilterFunctionSetter {
 }
 
 export const PullsProvider = function({children}: {children: React.ReactNode}) {
-   const [pulls, setFilter] = useFilteredPullsState();
-   return (<PullsContext.Provider value={{pulls, setFilter}}>
+   const unfilteredPulls = usePullsState();
+   const [filteredPulls, setFilter] = useFilteredPullsState(unfilteredPulls);
+   const sortedPulls = filteredPulls.sort(defaultSort);
+   return (<PullsContext.Provider value={{pulls: sortedPulls, setFilter}}>
       {children}
    </PullsContext.Provider>);
 }
+
+function defaultSort(a: Pull, b: Pull): number {
+   return (
+    // My pulls above pulls that aren't mine
+    compareBool(a.isMine(), b.isMine()) ||
+    // Pulls I have to CR/QA above those I don't
+    compareBool(a.hasOutdatedSig(getUser()), b.hasOutdatedSig(getUser())) ||
+    // Pulls I haven't touched vs those I have already CRed
+    compareBool(!a.hasCurrentSig(getUser()), !b.hasCurrentSig(getUser()))
```

Comment:
I wasn't confident that this worked, but I wrote up a spec in [Alloy](http://alloytools.org/) (attached below), and it looks like it should always satisfy the triangle inequality:

![image](https://user-images.githubusercontent.com/1283490/158707271-9b686fed-737e-40b7-a62c-9ffaf3e8dccc.png)

![image](https://user-images.githubusercontent.com/1283490/158707252-e74378bd-bed9-49be-a0bc-5671fcbdb3c7.png)


[pulldasher_sort_spec.zip](https://github.com/iFixit/pulldasher/files/8275773/pulldasher_sort_spec.zip)

Reply:
My intuition says "I can't imagine how it couldn't work".  The two equalities will still be 0 and the two inequalities will just reverse their values. Same affect as swapping the A for the B.